src/renderer/api-decorator.js

```diff
@@ -569,51 +569,43 @@ limitations under the License.
         const windowOptions = entityInfo.entityType === 'iframe' ? getWindowOptionsSync(entityInfo.parent) :
             getCurrentWindowOptionsSync();
 
-        let { plugin, preload } = convertOptionsToElectronSync(windowOptions);
+        let { preload } = convertOptionsToElectronSync(windowOptions);
 
-        if (plugin.length) {
-            evalPlugins(identity, plugin);
-        }
+        evalPlugins(identity);
 
         if (preload.length) {
             evalPreloadScripts(identity, preload);
         }
     });
 
     /**
-     * Requests plugin contents from the Core and evals them in the current window
+     * Requests plugin modules from the Core and evals them in the current window
      */
-    function evalPlugins(identity, pluginOption) {
+    function evalPlugins(identity) {
         const action = 'set-window-plugin-state';
+        const plugins = syncApiCall('get-plugin-modules');
         let logBase = `[plugin] [${identity.uuid}]-[${identity.name}]: `;
-        let plugins;
-
-        try {
-            plugins = syncApiCall('get-selected-preload-scripts', pluginOption);
-        } catch (error) {
-            return syncApiCall('write-to-log', { level: 'error', message: logBase + error });
-        }
 
         plugins.forEach((plugin) => {
-            const { name, version, content } = plugin;
+            const { name, version, _content } = plugin;
```

Comment:
why the `_`?

Reply:
This was done to separate plugin options. So `_content` will have the content of plugin modules, the rest of the options are original, whatever was set by the user in the manifest. The original options are needed here for logging purposes

Reply:
`options` here are the value `plugin` on line 590?

Reply:
Yeah so `plugin` here is the object that has original plugin options from the manifest + `_content` which has text of the module to be eval'ed

Reply:
this could be clarified a bit, not sure I could tell by just reading it.